### DISCUSSION TITLE
AR-Net Future Iteration Method Produces Autocorrelation Issues, Historic Metrics Decline

### DISCUSSION THREAD

**Discussion by mmangione** [ourownstory/neural_prophet#1519] [Q&A - get help using NeuralProphet]
We use NP for a number of prediction tasks and I've been focusing on improving the accuracy of our forecasts. One problem that I've been running into is that the shift-method the NP team uses to iteratively produce future predictions is both undocumented, undiscussed, and unexamined. Can someone from the NP team explain it to me?

For context here is the result of the training fit:
![Screenshot 2024-01-24 at 11 44 02 AM](https://github.com/ourownstory/neural_prophet/assets/6395900/75c1164a-6321-40e0-8b1b-94104c6d3f3e)

Here is the result of the 30 day forecast fit:
![Screenshot 2024-01-24 at 11 31 04 AM](https://github.com/ourownstory/neural_prophet/assets/6395900/f3d7b497-af69-4ed1-9c25-02407cb2cffd)

As I progress forward in my future predictions, the previous performance suffers. The historic fits to the data become less and less well-fitting, and every metric seems to suffer. I first noticed this in the uncertainty measurements. I have been using CQR and I notice that my miscoverage rate linearly increases with yhat.

![Untitled](https://github.com/ourownstory/neural_prophet/assets/6395900/9d3e0b42-0e5e-4dd8-a884-4c4496cc60ad)

So, I began to examine my results and found that it is no fluke. Each successive yhat has poorer performance on it's own historic data. So, I started looking at the residuals, and that's when I saw an indicator as to what is happening... whatever operation is being performed on the data is introducing autocorrelation issues. Very significant ones too.

In one particular dataset that was daily data with 1.5 years of history and a 30 day forward prediction window, yhat1 had:

- Dubin-Watson statistic of 1.897
- Ljung-Box statistic (with 1 lag) of

```
	lb_stat	lb_pvalue
1	1.399921	0.236737
```

![Screenshot 2024-01-24 at 11 27 26 AM](https://github.com/ourownstory/neural_prophet/assets/6395900/8fe08347-f310-41f9-9fa4-ddba3f815bbd)

However, when I looked at yhat30, it had:
- Durbin-Watson statistic of 0.844
- Ljung-Box statistic (with 1 lag) of

```
	lb_stat	lb_pvalue
1	184.792187	4.35678e-42
```

![Screenshot 2024-01-24 at 11 27 44 AM](https://github.com/ourownstory/neural_prophet/assets/6395900/340d5040-7811-48c5-8a9f-685847a2d15a)

For yhat1, the autocorrelation falls with in the acceptable parameters. For yhat30, the autocorrelation is very pronounced. DW says we've introduced positive autocorrelation errors, and LB just says it's displaying significant issues - on it's own historic prediction data. 

TLDR: Whatever is being done to produce these future predictions is introducing autocorrelation issues, does not maintain consistency for ar-lag numbers, and seems to result in linearly increasing errors for the same historical data. These are pretty significant results too.

**Comment by mmangione** [comment]
One element that I would expect to be somewhat close to constant over the course of a prediction cycle is the AR-lag number distribution. However, we are not seeing that in our data.

In the example above, here is the distribution of the AR-lag numbers for the training run:
![Screenshot 2024-01-24 at 11 32 08 AM](https://github.com/ourownstory/neural_prophet/assets/6395900/3c3e546c-f518-4b06-8bc3-582b3a6ecb77)

But, when I ask for a forecast 30 days in the future, here is the distribution of AR-lag numbers:
![Screenshot 2024-01-24 at 11 32 22 AM](https://github.com/ourownstory/neural_prophet/assets/6395900/2f3a07a6-c307-4f80-9c38-9c3d009a5c81)

I would expect that these values would be roughly similar, but they aren't even close. How can that be?

**Comment by ourownstory** [comment]
duplicate of #1546
